### PR TITLE
Allow referencing links in nested models in LiftDrag

### LINKED ISSUE #479: Support SDFormat 1.8 composition
SDFormat 1.8 has introduced new features regarding model composition as detailed in the [proposal](http://sdformat.org/tutorials?tut=composition_proposal&cat=pose_semantics_docs&). These include
  * Nested references to frames can be used in `relative_to`, `attached_to`, `expressed_in` attributes.
  * Frames can be used in the `canonical_link` attribute. The referenced frame can be nested
  * Frames can be used in the `//joint/parent` and `//joint/child`. The referenced frame can be nested
  * Models have a new attribute, [`placement_frame`](http://sdformat.org/tutorials?tut=composition_proposal&cat=pose_semantics_docs#1-4-4-placement-frame-model-placement_frame-and-include-placement_frame)
  * Models nested via `//include` are no longer [flattened](http://sdformat.org/tutorials?tut=composition&ver=1.5&#libsdformats-implementation-of-include-in-models) and work as though the included model was directly nested in the parent model.

Sub tasks:
  - [x] Support loading models nested via include
  - [x] Support nested references to frames
  -  ~~Support `canonical_link` attributes that use frames~~ This is currently not supported in SDFormat.
  - [x] Support the use of frames for a joint's parent or child
  - [x] Placement frames
  - [x] SDFormat generation/saving worlds with nested models and references
      - Included nested models end up being expanded even though the option "Expand include tags" is disabled
  - [x] Support Log recording and playback of worlds that contain nested models
  - [x] Support the use of nested references by Sensors
  - [x] Ensure default sensor topics are valid when nested models are used
  - [x] Ensure AABBs take nested models into account
  - [x] Support the use of GUI selection/transform tool with nested models
  - [ ] Support the use of nested references by ign-gazebo Systems
  - [x] Ensure that levels work with nested models

Related
- https://github.com/osrf/sdformat/issues/278

### PR DESCRIPTION
**Pull request by azeey**
# 🎉 New feature

Closes #479 

## Summary
This updates the `LiftDrag` system so that links in nested models can be referenced in the `link_name` parameter. This an example of using the `entitiesFromScopedName` function in `Util.hh`.

## Test it
Run the `INTEGRATION_lift_drag` test.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

File: test/integration/lift_drag_system.cc [hunk 1]
```diff
@@ -59,14 +73,16 @@ class LiftDragTestFixture : public ::testing::Test
 
 /////////////////////////////////////////////////
 /// Measure / verify force torques against analytical answers.
-TEST_F(LiftDragTestFixture, VerifyVerticalForce)
+TEST_P(VerticalForceParamFixture, VerifyVerticalForce)
 {
   using namespace std::chrono_literals;
+  ignition::common::setenv("IGN_GAZEBO_RESOURCE_PATH",
+                           PROJECT_SOURCE_PATH "/test/worlds/models");
```
Comment (jennuine) [inline test/integration/lift_drag_system.cc:80]: How about `common::joinPaths` for windows? (Below `sdfFile` as well)

File: src/systems/lift_drag/LiftDrag.hh [hunk 2]
```diff
@@ -38,7 +38,8 @@ namespace systems
   /// The following parameters are used by the system:
   ///
   /// link_name   : Name of the link affected by the group of lift/drag
-  ///               properties.
+  ///               properties. This can be a scoped name to reference links in
```
Comment (jennuine) [inline src/systems/lift_drag/LiftDrag.hh:41]: What do you think about adding `\sa entitiesFromScopedName` for users who do not know what scoped names are?

File: examples/worlds/lift_drag_nested.sdf [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+                <size>0.2 1.0 0.01</size>
+              </box>
+            </geometry>
+            <material>
+              <ambient>0.2 0.5 0.2 1.0</ambient>
+              <diffuse>.421 0.225 0.0 1.0</diffuse>
+            </material>
+          </visual>
+        </link>
+    </model>
```
Comment (jennuine) [inline examples/worlds/lift_drag_nested.sdf:222]: nit: weird indent

**Review by jennuine — COMMENTED** [review]
Looks good, left some minor comments

File: examples/worlds/lift_drag_nested.sdf [hunk 4]
```diff
@@ -0,0 +1,293 @@
+<?xml version="1.0" ?>
+<!--
+  This the same example as lift_drag.sdf but modified to show that links in
```
[the remark this answers is not in the document]
Comment (azeey) [reply examples/worlds/lift_drag_nested.sdf]: Done. https://github.com/ignitionrobotics/ign-gazebo/pull/955/commits/aa26825053a65c20700291b9b988ff12de37704e

File: src/systems/lift_drag/LiftDrag.cc [hunk 5]
```diff
@@ -172,33 +173,67 @@ void LiftDragPrivate::Load(const EntityComponentManager &_ecm,
...
   }
   else
   {
+    ignerr << "The LiftDrag system requires the 'link_name' parameter\n";
     this->validConfig = false;
+    return;
   }
 
 
   if (_sdf->HasElement("control_joint_name"))
```
[the remark this answers is not in the document]
Comment (azeey) [reply src/systems/lift_drag/LiftDrag.cc:210]: I copied the documentation from the variable `controlJointEntity` in LiftDrag.cc https://github.com/ignitionrobotics/ign-gazebo/pull/955/commits/aa26825053a65c20700291b9b988ff12de37704e

File: test/integration/lift_drag_system.cc (hunk 1 above)
Comment (azeey) [reply test/integration/lift_drag_system.cc]: Done https://github.com/ignitionrobotics/ign-gazebo/pull/955/commits/aa26825053a65c20700291b9b988ff12de37704e

File: src/systems/lift_drag/LiftDrag.hh (hunk 2 above)
Comment (azeey) [reply src/systems/lift_drag/LiftDrag.hh:41]: Good idea. https://github.com/ignitionrobotics/ign-gazebo/pull/955/commits/aa26825053a65c20700291b9b988ff12de37704e

File: examples/worlds/lift_drag_nested.sdf (hunk 3 above)
Comment (azeey) [reply examples/worlds/lift_drag_nested.sdf]: Fixed in https://github.com/ignitionrobotics/ign-gazebo/pull/955/commits/aa26825053a65c20700291b9b988ff12de37704e

**Comment by jennuine** [comment]
@osrf-jenkins run tests please

**Review by jennuine — APPROVED** [review]

**Pull request merged**